nilearn/input_data/base_masker.py

```diff
@@ -38,6 +40,40 @@ def filter_and_mask(imgs, mask_img_,
 
     mask_img_ = _utils.check_niimg(mask_img_, ensure_3d=True)
     imgs = _utils.check_niimgs(imgs, accept_3d=True)
+    if mask_time is None:
+        imgs = _utils.check_niimgs(imgs, accept_3d=True)
+    else:
+        # check type and value of mask_time
+        if (not isinstance(mask_time, list) or
+            len(mask_time) == 0 or
+            not isinstance(mask_time[0], bool)):
```

Comment:
A practical use case would be sub-selection as part of some sklearn-pipeline. Using this PR, memory is safed by avoiding creation of a tempory (sliced) nifti to pass on to the masker (index is provided at the time of masker instance creation). Just having idexing/slicing from Loic's PR cannot avoid this in a pipeline setting. ...I think.